File: src/main/java/com/trustline/trustline/appuser/service/EmailService.java
package com.trustline.trustline.appuser.service;

import com.trustline.trustline.appuser.dto.EmailRequest;
import com.trustline.trustline.appuser.model.OtpModeEnum;
import com.trustline.trustline.appuser.model.VerificationModel;
import com.trustline.trustline.appuser.model.VerificationType;

import java.util.Optional;
import java.util.UUID;

public interface EmailService {
    String sendMail(EmailRequest emailRequest);
    VerificationModel saveVerification(OtpModeEnum mode, String messageId, UUID userId, String otp, VerificationType type);
    Optional<VerificationModel> getVerificationById(UUID id);
    Optional<VerificationModel> getbyUserIdAndPin(UUID userId, String pin);
}

File: src/main/kotlin/trustline/appuser/service/EmailService.kt
package trustline.appuser.service;

import trustline.appuser.dto.EmailRequest;
import trustline.appuser.dto.OtpModeEnum
import trustline.appuser.dto.Status
import trustline.appuser.dto.VerificationType
import trustline.appuser.model.UserModel
import trustline.notification.model.VerificationModel
import java.util.*

interface EmailService {
    fun sendMail(emailRequest: EmailRequest): String?
    fun saveVerification(
        mode: OtpModeEnum,
        messageId: String,
        user: UserModel,
        otp: String,
        type: VerificationType,
        status: Status
    ): VerificationModel?

    fun getVerificationById(id: UUID): Optional<VerificationModel>
    fun getVerificationByIdAndStatus(id: UUID, status: Status): VerificationModel?
    fun getbyUserIdAndPinAndStatus(userId: UUID, pin: String, status: Status): VerificationModel?
    fun updateVerification(verificationModel: VerificationModel): VerificationModel?
}
